docs/blog-posts/symmetric-key-support.md

```diff
@@ -0,0 +1,220 @@
+- The security key is stored in the CachedValue property of CredentialDescription
+- This design maintains independence from Microsoft.IdentityModel types in the abstractions layer
+- The implementation follows the same pattern as certificate handling for consistency
+
+## Design
+
+### 1. New CredentialSource Values(Abstractions Layer)
+```csharp
+public enum CredentialSource
+{
+    // Existing values
+    Certificate = 0,
+    KeyVault = 1,
+    Base64Encoded = 2,
+    Path = 3,
+    StoreWithThumbprint = 4,
+    StoreWithDistinguishedName = 5,
+
+    // New values
+    SymmetricKeyFromKeyVault = 6,
+    SymmetricKeyBase64Encoded = 7
+}
+```
+
+### 2. SymmetricKeyDescription Class(IdWeb Layer)
+Following the same pattern as CertificateDescription:
+
+```csharp
+public class SymmetricKeyDescription : CredentialDescription
+{
+    public static SymmetricKeyDescription FromKeyVault(string keyVaultUrl, string secretName)
+    {
+        return new SymmetricKeyDescription
+        {
+            SourceType = CredentialSource.SymmetricKeyFromKeyVault,
+            KeyVaultUrl = keyVaultUrl,
+            KeyVaultSecretName = secretName
+        };
+    }
+
+    public static SymmetricKeyDescription FromBase64Encoded(string base64EncodedValue)
+    {
+        return new SymmetricKeyDescription
+        {
+            SourceType = CredentialSource.SymmetricKeyBase64Encoded,
+            Base64EncodedValue = base64EncodedValue
+        };
+    }
+}
+```
+
+### 3. New Loader Classes(IdWeb Layer)
+Internal implementation in Microsoft.Identity.Web:
+
+```csharp
+internal class KeyVaultSymmetricKeyLoader : ICredentialSourceLoader
+{
+    private readonly SecretClient _secretClient;
+
+    public KeyVaultSymmetricKeyLoader(SecretClient secretClient)
+    {
+        _secretClient = secretClient ?? throw new ArgumentNullException(nameof(secretClient));
+    }
+
+    public async Task LoadIfNeededAsync(CredentialDescription description, CredentialSourceLoaderParameters? parameters)
+    {
+        _ = Throws.IfNull(description);
+
+        if (description.CachedValue != null)
+            return;
+
+        if (string.IsNullOrEmpty(description.KeyVaultUrl))
+            throw new ArgumentException("KeyVaultUrl is required for KeyVault source");
+
+        if (string.IsNullOrEmpty(description.KeyVaultSecretName))
+            throw new ArgumentException("KeyVaultSecretName is required for KeyVault source");
+
+        // Load secret from Key Vault
+        var secret = await _secretClient.GetSecretAsync(description.KeyVaultSecretName).ConfigureAwait(false);
+        if (secret?.Value == null)
+            throw new InvalidOperationException($"Secret {description.KeyVaultSecretName} not found in Key Vault");
+
+        try
+        {
+            // Convert secret value to bytes and create SymmetricSecurityKey
+            var keyBytes = Convert.FromBase64String(secret.Value.Value);
+            description.CachedValue = new SymmetricSecurityKey(keyBytes);
```

Comment:
The fact that `CachedValue` is an object is ... unpleasant. It requires users to cast to the right object (string, X509Certificate2 etc.). 

At least for certificates, a `Certificate` property also exists to avoid a cast. I suggest you do the same for symmetric keys?

Reply:
(read my comments below about extending CredentialDescription first)